mlp.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Optional, Tuple
+
+import torch
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import BCELoss, Linear, Sigmoid
+from torch.optim import AdamW
+
+
+class LitLinear(LightningModule):
+    def __init__(self, in_features: int, out_features: int) -> None:
+        super().__init__()
+        self.linear = Linear(in_features=in_features, out_features=out_features)
+        self.sigmoid = Sigmoid()
+        self.bce = BCELoss()
+
+    def forward(self, x):
+        x = self.linear(x)
+        out = self.sigmoid(x)
+        return out
```

Comment:
So torch has a really handy [`BCEWithLogitsLoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html) that combines the sigmoid and `BCELoss` into one operation that is more numerically-stable than the two individually. I would strongly recommend to use that.

```suggestion
        self.loss = BCEWithLogitsLoss()

    def forward(self, x):
        x = self.linear(x)
        out = self.loss(x)
        return out
```